app/commands/create_grade_year.rb

```diff
@@ -0,0 +1,25 @@
+class CreateGradeYear < ROM::Commands::Create[:sql]
+  relation :grade_years
+  register_as :create
+  result :one
+
+  def execute(attributes)
+    gpa = attributes.delete(:gpa).to_f
+    if gpa > 3
+      letter = 'A'
+    elsif gpa > 1
+      letter = 'D'
+    end
+
+    grade = Grade.find_by_letter(letter)
+    grade_year = GradeYear.create!(
+      student_id: attributes[:student_id],
+      gpa: gpa,
+      grade_id: grade.id,
+      year: attributes[:year]
+    )
```

Comment:
I'd really like to start using ROM at work but in order to do that I'll have to get buy in from my coworkers and to do that the transition is going to have to be incremental.

ROM has a great paradigm but one that is unfamiliar to most rails devs. This first pass is a way to start separating code. Then in the next step I'd like to start replacing the AR stuff with ROM relations and POROs.

So for this command I imagine that figuring out of gpa to grade would be one component and then the creation of the actual grade year would be done by a relation. Still have to figure out how to do that though :)